packages/extension-utils/src/OAuthConfigForm/OAuthConfigForm.tsx

```diff
@@ -176,107 +131,111 @@ export const OAuthConfigForm = ({
       client_id: clientId,
       redirect_uri,
     }
-    updateFields({
-      [BASE_URL]: baseUrl,
-      [WEB_URL]: webUrl,
-    })
     // TODO: replace when redux is introduced
     localStorage.setItem(configKey, JSON.stringify(data))
     if (setHasConfig) setHasConfig(true)
     setSaved(data)
-    updateMessage(POSITIVE, `Saved ${webUrl} as OAuth server`)
+    updateMessageBar('positive', `Saved ${webUrl} as OAuth server`)
   }
 
-  const updateForm = async (_e: BaseSyntheticEvent, save: boolean) => {
-    updateMessage('inform', '')
-    const versionsUrl = `${fields.baseUrl}/versions`
+  const verifyUrl = async (): Promise<ILookerVersions | undefined> => {
+    updateMessageBar(messageBarIntent, '')
+    const versionsUrl = `${apiServerUrlValue}/versions`
     try {
-      updateFields({
-        fetchedUrl: fields.baseUrl,
-      })
-      const { web_server_url: webUrl, api_server_url: baseUrl } =
-        (await getVersions(versionsUrl)) as ILookerVersions
-      updateMessage(POSITIVE, 'Configuration is valid')
-      updateFields({
-        [BASE_URL]: baseUrl,
-        [WEB_URL]: webUrl,
-      })
-      if (save) {
-        saveConfig(baseUrl, webUrl)
+      setFetchedUrl(apiServerUrlValue)
+      const versions = await getVersions(versionsUrl)
+
+      if (versions) {
+        updateMessageBar('positive', 'Configuration is valid')
+        setWebUrlValue(versions.web_server_url)
       }
+
+      return versions
     } catch (e: any) {
       fetchError(e.message)
+      return undefined
     }
   }
 
-  const handleSave = async (e: BaseSyntheticEvent) => {
-    await updateForm(e, true)
+  const handleSaveClick = async () => {
+    const versions = await verifyUrl()
+    if (versions) {
+      saveConfig(versions.api_server_url, versions.web_server_url)
+    }
   }
 
-  const handleVerify = async (e: BaseSyntheticEvent) => {
-    await updateForm(e, false)
+  const handleVerifyClick = () => {
```

Comment:
I prefer the previous handler names, I find them more concise. `handleVerify`, `handleSave`, `handleClear`, `handleLogin` etc.

Reply:
agreed. The verb already makes the method name clear